### PR TITLE
winpr: add some ASN1 parsing functions

### PR DESCRIPTION
**Pull request by hardening**
This patch implements ASN1 parsing in WinPR, as we need it in SSPI packages.

### CODE REVIEW

File: winpr/include/winpr/asn1.h [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+#define ER_TAG_BIT_STRING 0x03
+#define ER_TAG_OCTET_STRING 0x04
+#define ER_TAG_NULL 0x05
+#define ER_TAG_OBJECT_IDENTIFIER 0x06
+#define ER_TAG_ENUMERATED 0x0A
+#define ER_TAG_UTF8STRING 0x0C
+#define ER_TAG_PRINTABLE_STRING 0x13
+#define ER_TAG_IA5STRING 0x16
+#define ER_TAG_UTCTIME 0x17
+#define ER_TAG_GENERAL_STRING 0x1B
```
Comment (akallabeth) [inline winpr/include/winpr/asn1.h:39]: maybe enum the tags, if they are unique?

File: winpr/include/winpr/stream.h [hunk 2]
```diff
@@ -287,6 +287,15 @@ extern "C"
 		*_s->pointer++ = (_v)&0xFF;
 	}
 
+    static INLINE void Stream_Write_UINT24_BE(wStream* _s, UINT16 _v)
```
Comment (akallabeth) [inline winpr/include/winpr/stream.h:290]: sure this should be `UINT16` and not `UINT32`?

Comment (hardening) [reply winpr/include/winpr/stream.h:290]: good catch

File: winpr/libwinpr/utils/test/TestASN1.c [hunk 3]
```diff
@@ -0,0 +1,288 @@
...
+        return -26;
+
+    if (!WinPrAsn1DecReadContextualInteger(&decoder, 0, &error, &integerV))
+        return -27;
+
+
+    return 0;
+}
+
+static BYTE oid1_val[] = { 1 };
```
Comment (akallabeth) [inline winpr/libwinpr/utils/test/TestASN1.c:141]: better make these `const` if they are not to be modified

**Comment by akallabeth** [comment]
@freerdp-bot test

**Review by akallabeth — APPROVED** [review]
LGTM

**Pull request merged**